interface/admin.py

```diff
@@ -1,6 +1,49 @@
+from zipfile import ZipFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
 from django.contrib import admin
+from django.db.models import F
+from django.http import FileResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, _assignment):
+        if _assignment.count() != 1:
```

Comment:
Prefixing an argument with `_` usually means it's ignored/irrelevant. You can name this one `queryset`, it's a good convention for admin actions.